underfs/abfs/src/main/java/alluxio/underfs/abfs/AbfsUnderFileSystem.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.abfs;
+
+import alluxio.AlluxioURI;
+import alluxio.conf.PropertyKey;
+import alluxio.underfs.UfsFileStatus;
+import alluxio.underfs.UfsStatus;
+import alluxio.underfs.UnderFileSystemConfiguration;
+import alluxio.underfs.hdfs.HdfsUnderFileSystem;
+import alluxio.underfs.options.FileLocationOptions;
+
+import com.google.common.base.MoreObjects;
+import org.apache.hadoop.conf.Configuration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A Microsoft Azure Data Lake Storage Gen2 Implementation.
+ */
+@ThreadSafe
+public class AbfsUnderFileSystem extends HdfsUnderFileSystem {
+  private static final Logger LOG = LoggerFactory.getLogger(AbfsUnderFileSystem.class);
+
+  /** Constant for the abfs URI scheme. */
+  public static final String SCHEME_INSECURE = "abfs://";
+
+  /** Constant for the abfss URI scheme. */
+  public static final String SCHEME_SECURE = "abfss://";
```

Comment:
Is there any additional configuration needed for connecting with secured scheme?

Reply:
https://hadoop.apache.org/docs/current/hadoop-azure/abfs.html#Security 

Has not indicated otherwise. So doesn't seem to be the case.

Reply:
➜  alluxio git:(azure_gen2) ./bin/alluxio fs mount \
  --option fs.azure.account.key.bradleytest1.dfs.core.windows.net=<SHARED_KEY> \
  /abfs abfss://container@bradleytest1.dfs.core.windows.net/


Mounted abfss://container@bradleytest1.dfs.core.windows.net/ at /abfs